File: mobile-app/lib/domain/entities/events/event_status.dart
enum EventStatus {
  draft,
  open,
  ongoing,
  closed,
  cancelled,
  unknown,
}

extension EventStatusParser on EventStatus {
  static EventStatus fromApiValue(int value) {
    switch (value) {
      case 0:
        return EventStatus.draft;
      case 1:
        return EventStatus.open;
      case 2:
        return EventStatus.ongoing;
      case 3:
        return EventStatus.closed;
      case 4:
        return EventStatus.cancelled;
      default:
        return EventStatus.unknown;
    }
  }
}

File: backend-api/UniYouth.Api/Shared/Enums/EventStatus.cs
namespace UniYouth.Api.Shared.Enums
{
    /// <summary>
    /// Trạng thái sự kiện
    /// </summary>
    public enum EventStatus : byte
    {
        Draft = 0,        // Nháp – chưa công bố
        Open = 1,         // Mở đăng ký
        Ongoing = 2,      // Đang diễn ra
        Closed = 3,       // Đã kết thúc
        Cancelled = 4     // Đã hủy
    }
}
